Library/Homebrew/extend/io.rb

```diff
@@ -1,22 +1,24 @@
-# typed: false
+# typed: true
 # frozen_string_literal: true
 
 class IO
   def readline_nonblock(sep = $INPUT_RECORD_SEPARATOR)
     line = +""
     buffer = +""
```

Comment:
this needs to be outside the `rescue`d block, so that `line` is provably a `String` when `line.empty?` is invoked below.